docs/widget-layouts.md

```diff
@@ -0,0 +1,61 @@
+# Widget layout support
+
+Rooms can have a default widget layout to auto-pin certain widgets, make the container different
+sizes, etc. These are defined through the `io.element.widgets.layout` state event (empty state key).
+
+Full example content:
+```json5
+{
+  "widgets": {
+    "first-widget-id": {
+      "container": "top",
+      "index": 0,
+      "width": 60,
+      "height": 40
+    },
+    "second-widget-id": {
+      "container": "right"
+    }
+  }
+}
+```
+
+As shown, there are two containers possible for widgets. These containers have different behaviour
+and interpret the other options differently. 
+
+## `top` container
+
+This is the "App Drawer" or any pinned widgets in a room. This is by far the most versatile container
+though does introduce potential usability issues upon members of the room (widgets take up space and
+therefore less messages can be shown).
+
+The `index` for a widget determines which order the widgets show up in from left to right. Widgets
+without an `index` will show up as the rightmost widgets. Tiebreaks (same `index` or multiple defined
+without an `index`) are resolved by comparing widget IDs. A maximum of 3 widgets can be in the top
+container - any which exceed this will be ignored. Smaller numbers represent leftmost widgets.
+
+The `width` is relative width within the container in percentage points. This will be clamped to a
```

Comment:
I wonder if a system like the `flex` CSS property would be better, where the numbers are just "ratios" rather than percentages? So, if widget 1 has "width: 1" and widget 2 has "width: 2", it means widget 2 is twice as wide as widget 1. Since there are two widgets in this example, it means widget 1 will get 1/3 of available space, while widget 2 will get 2/3.

Maybe it falls apart though because we already allow dragging to adjust layout to arbitrary points, which would encode some quite strange ratios... 😅 Anyway, seemed worth consideration at least.

Reply:
Is be worried about values getting obscenely large or out of hand with ratios - we can't store floats.

Percentages also feel somewhat more predictable, imo.

Reply:
Okay, I think I just find the percentages distasteful in some way, like they are too low level or something, but I don't think it's worth holding up progress for that, so feel free to continue as you like.

Reply:
Aside from feelings about percentages vs other units of measure, there's also time pressure, external requirements, etc pushing this into needing to be percentages :( 

We can revist this one day - it is deliberately in our namespace to let us make changes as needed.